content/5_Plat/Centroid.mdx

```diff
@@ -86,6 +90,70 @@ int main() {
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class findCentroid {
+	public static int[] subSize;
+	public static List<Integer>[] adj;
+	public static int N;
+
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+		N = io.nextInt();
+		adj = new List[N];
+		for (int i = 0; i < N; i++) {
+			adj[i] = new ArrayList<>();
+		}
+		subSize = new int[N];
+
+		for (int i = 0; i < N - 1; i++) {
+			int a = io.nextInt() - 1;
+			int b = io.nextInt() - 1;
+			adj[a].add(b);
+			adj[b].add(a);
+		}
+
+		subtreeSize(0, -1);
+		io.println(getCentroid(0, -1) + 1);
+		io.close();
+
+	}
+
+	// Find the size of the subtree under this node.
+	public static int subtreeSize(int node, int par) {
+		int res = 1;
+		for (int next : adj[node]) {
+			if (next == par) {
+				continue;
+			}
+			res += subtreeSize(next, node);
+		}
+		return (subSize[node] = res);
+	}
+
+	// Find the centroid of the tree (the subtree with <= N/2 nodes)
+	public static int getCentroid(int node, int par) {
+		for (int next : adj[node]) {
+			if (next == par) {
+				continue;
+			}
+			// Keep searching for the centroid if there are subtrees with more than N/2 nodes.
+			if (subSize[next] * 2 > N) {
+				return getCentroid(next, node);
+			}
+		}
+		return node;
+	}
+	// Codesnip{Kattio}
```

Comment:
you sure this works? i thought it was `//CodeSnip{Kattio}` w/o the space in the middle